src/Projection/InMemoryEventStoreQuery.php

```diff
@@ -12,16 +12,345 @@
 
 namespace Prooph\EventStore\Projection;
 
+use Closure;
+use Iterator;
+use Prooph\Common\Messaging\Message;
+use Prooph\EventStore\ActionEventEmitterEventStore;
+use Prooph\EventStore\EventStore;
+use Prooph\EventStore\Exception;
 use Prooph\EventStore\InMemoryEventStore;
+use Prooph\EventStore\StreamName;
 
-final class InMemoryEventStoreQuery extends AbstractQuery
+final class InMemoryEventStoreQuery implements Query
 {
-    use InMemoryEventStoreQueryTrait;
+    /**
+     * @var array
+     */
+    private $knownStreams;
 
-    public function __construct(InMemoryEventStore $eventStore)
+    /**
+     * @var EventStore
+     */
+    private $eventStore;
+
+    /**
+     * @var array
+     */
+    private $streamPositions;
+
+    /**
+     * @var array
+     */
+    private $state = [];
+
+    /**
+     * @var callable|null
+     */
+    private $initCallback;
+
+    /**
+     * @var Closure|null
+     */
+    private $handler;
+
+    /**
+     * @var array
+     */
+    private $handlers = [];
+
+    /**
+     * @var boolean
+     */
+    private $isStopped = false;
+
+    /**
+     * @var ?string
+     */
+    private $currentStreamName = null;
+
+    public function __construct(EventStore $eventStore)
+    {
+        $this->eventStore = $eventStore;
+
+        if ($eventStore instanceof ActionEventEmitterEventStore) {
```

Comment:
I'm with @oqq (see https://github.com/prooph/event-store/issues/240#issuecomment-270326292), that a `EventStoreDecorator` interface would be useful for this scenario here.
Maybe a user wants to use a custom event store decorator in this case the factory would not detect it.

Thoughts? we can add this interface in a separate PR. This one is already large enough :D